src/CSnakes.Runtime/ProcessUtils.cs

```diff
@@ -1,45 +1,41 @@
 using CSnakes.Runtime.Locators;
 using Microsoft.Extensions.Logging;
+using System.Collections.ObjectModel;
 using System.Diagnostics;
 
 namespace CSnakes.Runtime;
 
 internal static class ProcessUtils
 {
-    internal static (Process proc, string? result, string? errors) ExecutePythonCommand(ILogger? logger, PythonLocationMetadata pythonLocation, string arguments)
+    internal static (Process proc, string? result, string? errors) ExecutePythonCommand(ILogger? logger, PythonLocationMetadata pythonLocation, params string[] arguments)
```

Comment:
Consider `ReadOnlySpan<string>` for the variadic `arguments` ([which is possible since C#13](https://devblogs.microsoft.com/dotnet/csharp13-calling-methods-is-easier-and-faster/)) to potentially save on array allocation.

If you also introduce a generic extension like this:

```c#
private static void AddTo<T>(this ReadOnlySpan<T> items, ICollection<T> collection)
{
    foreach (var item in items)
        collection.Add(item);
}
```

the adding `arguments` elements to `ArgumentList` of `ProcessStartInfo` would be simple as:

```c#
arguments.AddTo(startInfo.ArgumentList);
```

Reply:
These commands are run so rarely and take so long I don't think it'll make a noticable difference?